src/cdk-experimental/scrolling/auto-size-virtual-scroll.ts

```diff
@@ -192,11 +197,12 @@ export class AutoSizeVirtualScrollStrategy implements VirtualScrollStrategy {
         const overscan = (scrollDelta < 0 ? endBuffer : startBuffer) - this._minBufferPx +
             scrollMagnitude;
         // The number of currently rendered items to remove on the side the user is scrolling away
-        // from.
-        const removeItems = Math.max(0, Math.floor(overscan / this._averager.getAverageItemSize()));
+        // from. If removal has failed in recent cycles we are less aggressive in how much we try to
+        // remove.
+        const removeItems = Math.min(renderedRange.end - renderedRange.start, Math.max(0,
```

Comment:
This expression is getting pretty long, consider moving out the `Math.floor` into a variable?